repos/system_upgrade/el7toel8/actors/systemfacts/libraries/systemfacts.py

```diff
@@ -206,25 +207,23 @@ def _parse(r):
 
 def get_selinux_status():
     ''' Get SELinux status information '''
-    def asbool(x):
-        return x == 'enabled'
-
-    def identity(x):
-        return x
-
-    key_mapping = (('SELinux status', 'enabled', asbool),
-                   ('Policy MLS status', 'mls_enabled', asbool),
-                   ('Loaded policy name', 'policy', identity),
-                   ('Current mode', 'runtime_mode', identity),
-                   ('Mode from config file', 'static_mode', identity),)
 
     outdata = {}
-    for item in call(['sestatus']):
-        for (key, mapped, cast) in key_mapping:
-            if item.startswith(key):
-                outdata[mapped] = cast(item[len(key)+1:].strip())
+    try:
+        import selinux
+        outdata['enabled'] = True if selinux.is_selinux_enabled() == 1 else False
+        outdata['mls_enabled'] = True if selinux.is_selinux_mls_enabled == 1 else False
+        outdata['runtime_mode'] = "enforcing" if selinux.security_getenforce() == 1 else "permissive"
+        enforce_mode = selinux.selinux_getenforcemode()
+        if enforce_mode >= 0:
+            outdata['static_mode'] = "enforcing" if enforce_mode == 1 else "permissive"
+        else:
+            outdata['static_mode'] = "disabled"
+        outdata['policy'] = selinux.selinux_getpolicytype()[1]
 
-    return SELinux(**outdata)
+        return SELinux(**outdata)
+    except ImportError as error:
+        api.report_error("SELinux Import Error", details="libselinux-python package must be installed.")
```

Comment:
api.report_error will keep failing till leapp #405 is not merged into the master.